benches/setup.rs

```diff
@@ -0,0 +1,108 @@
+use std::fs;
+use std::path::PathBuf;
+
+use log::info;
+
+use dapol::accumulators::{NdmSmt, NdmSmtConfigBuilder};
+use dapol::read_write_utils;
+use dapol::{EntityId, Height, InclusionProof, MaxThreadCount};
+
+// CONSTANTS
+// ================================================================================================
+
+pub const TREE_HEIGHTS: [u8; 3] = [16, 32, 64];
+pub const NUM_USERS: [u64; 35] = [
+    10_000,
+    20_000,
+    30_000,
+    40_000,
+    50_000,
+    60_000,
+    70_000,
+    80_000,
+    90_000,
+    100_000,
+    200_000,
+    300_000,
+    400_000,
+    500_000,
+    600_000,
+    700_000,
+    800_000,
+    900_000,
+    1_000_000,
+    2_000_000,
+    3_000_000,
+    4_000_000,
+    5_000_000,
+    6_000_000,
+    7_000_000,
+    8_000_000,
+    9_000_000,
+    10_000_000,
+    30_000_000,
+    50_000_000,
+    70_000_000,
+    90_000_000,
+    100_000_000,
+    125_000_000,
+    250_000_000,
+];
+
+// HELPER FUNCTIONS
+// ================================================================================================
+
+pub fn build_ndm_smt(tup: (Height, MaxThreadCount, u64)) -> Result<NdmSmt, &'static str> {
+    let height_int = tup.0.as_raw_int();
+    let max_users_for_height = 2_u64.pow((height_int - 1) as u32);
+
+    if tup.2 as u64 > max_users_for_height {
+        return Err("Number of users exceeds maximum");
+    }
+
+    NdmSmtConfigBuilder::default()
+        .height(tup.0)
+        .max_thread_count(tup.1)
+        .num_entities(tup.2)
+        .secrets_file_path(PathBuf::from("examples/ndm_smt_secrets_example.toml"))
+        .build()
+        .map_err(|_| "Unable to build NdmSmtConfig")?
```

Comment:
Does the logging work when running benches? I'm pretty sure errors are logged but if the benches don't support logging or the logging is not turned on then this will swallow the error message and so debugging anything will be hard.

Reply:
I'll investigate

Reply:
See above